### DISCUSSION TITLE
File transfer functionality for Server

### DISCUSSION THREAD

**Discussion by bwallner** [FreeOpcUa/opcua-asyncio#1552] [Q&A]
Hy, I am trying to use the asyncua package to set up a server which is able to transfer files. I have discovered, that this package has already some kind of solution for the client side - I am looking for the server counterpart. 

I tried using the "FileType" Object-Type but it seems none of the provided methods (open, close, read, write, etc.) within the filetype are implemented. Is this the case or did I just misunderstand the functionality? If they are not implemented, what would be a proper way to implement those methods myself?

Best wishes and thanks for your effort!

**Comment by schroeder-** [comment]
There is no server functionality. One blocker is that there are no session informations avialable in the method context. This is needed store the file handles of a client, to isulate them from other clients and close them when a client disconnects.

**Reply by bwallner** [reply]
@schroeder- Thank you for the information!
is this functionality planned to be implemented? Since I am not very familiar with the source code of the library, where would be the entry point to understand the problem regarding the session information for the method context?
Should I create an issue /feature request?
Best wishes!
